scipy/special/tests/test_basic.py

```diff
@@ -1666,8 +1666,9 @@ def test_euler(self):
         eu0 = special.euler(0)
         eu1 = special.euler(1)
         eu2 = special.euler(2)   # just checking segfaults
-        assert_almost_equal(eu0[0],1,8)
-        assert_almost_equal(eu2[2],-1,8)
+        assert_allclose(eu0, [1], rtol=1e-15)
+        assert_allclose(eu1, [1, 0], rtol=1e-15)
```

Comment:
Just to clarify, certainly most of the time we lose relative tolerance around zeros of functions. But in this case all we have to do is check whether the input is odd and if it is return `0`, so IMO we should get it exactly right.

Looking at the specfun [source](https://github.com/scipy/scipy/blob/master/scipy/special/specfun/specfun.f#L2560) though it appears that the odd values don't get initialized--so the fact that we're getting it right now is probably just luck. (From the f2py source it doesn't look like arrays get initialized to `0`, though I could easily have missed something.)